src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,203 @@
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.alertRuleFetchErrorTitle', {
+          defaultMessage: 'Error fetching alert rule',
+        });
+        displayError(errorTitle, error.message, toastNotifications);
+      }
+    };
+
+    const fetchSearchSource = async (fetchedAlert: Alert<SearchThresholdAlertParams>) => {
+      try {
+        return await data.search.searchSource.create(fetchedAlert.params.searchConfiguration);
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.searchSourceErrorTitle', {
+          defaultMessage: 'Error fetching search source',
+        });
+        displayError(errorTitle, error.message, toastNotifications);
+      }
+    };
+
+    const showDataViewFetchError = (alertId: string) => {
+      const errorTitle = i18n.translate('discover.viewAlert.dataViewErrorTitle', {
+        defaultMessage: 'Error fetching data view',
+      });
+      displayError(
+        errorTitle,
+        new Error(`Data view failure of the alert rule with id ${alertId}.`).message,
+        toastNotifications
+      );
+    };
+
+    const navigateToResults = async () => {
+      const fetchedAlert = await fetchAlert();
+      if (!fetchedAlert) {
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      const calculatedChecksum = getCurrentChecksum(fetchedAlert.params);
+      if (openConcreteAlert && calculatedChecksum !== queryParams.checksum) {
+        displayRuleChangedWarn(toastNotifications);
+      } else if (openConcreteAlert && calculatedChecksum === queryParams.checksum) {
+        displayPossibleDocsDiffInfoAlert(toastNotifications);
+      }
+
+      const fetchedSearchSource = await fetchSearchSource(fetchedAlert);
+      if (!fetchedSearchSource) {
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      const dataView = fetchedSearchSource.getField('index');
+      const timeFieldName = dataView?.timeFieldName;
+      if (!dataView || !timeFieldName) {
+        showDataViewFetchError(fetchedAlert.id);
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      let timeRange: TimeRange;
+      if (openConcreteAlert) {
+        timeRange = { from: queryParams.from, to: queryParams.to };
+      } else {
+        const filter = getTime(dataView, {
+          from: `now-${fetchedAlert.params.timeWindowSize}${fetchedAlert.params.timeWindowUnit}`,
+          to: 'now',
+        });
+        timeRange = {
+          from: filter?.query.range[timeFieldName].gte,
+          to: filter?.query.range[timeFieldName].lte,
+        };
+      }
+
+      const state: DiscoverAppLocatorParams = {
+        query: fetchedSearchSource.getField('query') || data.query.queryString.getDefaultQuery(),
+        indexPatternId: dataView.id,
+        timeRange,
+      };
+      const filters = fetchedSearchSource.getField('filter');
+      if (filters) {
+        state.filters = filters as Filter[];
```

Comment:
why is the cast necessary?

Reply:
Since `getField` currently returns the [**following**](https://github.com/elastic/kibana/pull/123206#discussion_r798341472).